### DISCUSSION TITLE
Temporary failure in name resolution

### DISCUSSION THREAD

**Discussion by toumku** [community/community#46833] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I can't clone my own private repo in ubuntu 20.04.

My ~/.ssh/config file is

```
Host github-toumku
	HostName github.com
	User git
	IdentityFile ~/.ssh/id_github_toumku
```

If i try `ssh -T github-toumku`, says `Hi toumku! You've successfully authenticated, but GitHub does not provide shell access.`.
Then i try `git clone git@github-toumku:<org>/<repo>.git .`, it fails. I 100% sure i have full access, org name is correct, repo is correct. So it should not be wrong. I tried 3 times 2 different computer and they works without any issues.

```
ssh: Could not resolve hostname github-toumku: Temporary failure in name resolution
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

I can't even clone using HTTPS. Because: 
```
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/<org>/<repo>.git/'
```

I tried full day to just clone repo. But still cant. Things i tried:

- Restarted my server
- Running with sudo
- Many times checked pub key is added to Github SSH keys
- Added ssh-add to my private key
- Internet is works fine
- Cloned public repo (works using HTTPS)

Now i have no idea how to clone. PS: Sorry my english not good

Why not git@github.com?. Because i have two github account. I tested locally many times.

**Comment by Juanescacha** [comment]
let me see if I got you correctly, You have tried all the different ways to clone a repo to your machine but could'nt do it?
if that's the case I tried with the https and got no issue.

![image](https://user-images.githubusercontent.com/6394715/218265613-8e3ae244-4aee-4159-bb7c-0258724c350b.png)

maybe could be an issue git your configuration with git, so try erasing all git configuration to start from default

# Remove the global configuration file 
$ rm ~/.gitconfig

# Check if the file got actually eliminated
$ ls ~/.gitconfig
ls: cannot access /root/.gitconfig: No such file or directory

tell me if it helps you!

**Reply by epbonacina** [reply]
Removing my `.gitconfig` file worked for me. Thanks!

**Comment by lbirkert** [comment]
> ### Select Topic Area
> 
> Question
> ### Body
> 
> I can't clone my own private repo in ubuntu 20.04.
> 
> My ~/.ssh/config file is
> 
> ```
> Host github-toumku
> 	HostName github.com
> 	User git
> 	IdentityFile ~/.ssh/id_github_toumku
> ```
> 
> If i try `ssh -T github-toumku`, says `Hi toumku! You've successfully authenticated, but GitHub does not provide shell access.`. Then i try `git clone git@github-toumku:<org>/<repo>.git .`, it fails. I 100% sure i have full access, org name is correct, repo is correct. So it should not be wrong. I tried 3 times 2 different computer and they works without any issues.
> 
> ```
> ssh: Could not resolve hostname github-toumku: Temporary failure in name resolution
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
> ```
> 
> I can't even clone using HTTPS. Because:
> 
> ```
> remote: Support for password authentication was removed on August 13, 2021.
> remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
> fatal: Authentication failed for 'https://github.com/<org>/<repo>.git/'
> ```
> 
> I tried full day to just clone repo. But still cant. Things i tried:
> 
>     * Restarted my server
> 
>     * Running with sudo
> 
>     * Many times checked pub key is added to Github SSH keys
> 
>     * Added ssh-add to my private key
> 
>     * Internet is works fine
> 
>     * Cloned public repo (works using HTTPS)
> 
> 
> Now i have no idea how to clone. PS: Sorry my english not good
> 
> Why not [git@github.com](mailto:git@github.com)?. Because i have two github account. I tested locally many times.

You could try explicitly adding the `ssh://` prefix to the url. Why don't you add your id_rsa public key to both github accounts tho?

**Comment by toumku** [comment]
I cloned with https using personal access token. Thanks.

**Reply by Juanescacha** [reply]
Don't forget to mark the answer to help other if they get into same issue

**Reply by toumku** [reply]
> Don't forget to mark the answer to help other if they get into same issue

I think there is no exact answer. I fixed it in a different way. So i will not accept any answer. Am i right? I'm new.

**Comment by linushahs** [comment, marked as answer]
@toumku  @Juanescacha 
# cat /etc/resolv.conf
nameserver 8.8.4.4
nameserver 8.8.8.8

This will solve the problem. Change dns server as given above using any linux editor (nano or vim).

**Reply by dudvdw** [reply]
It works! Thanks~

**Reply by ALittleLessThanMediocre** [reply]
This is the correct answer, thanks!

**Reply by JRcxl** [reply]
useful
